frontend/src/metabase/dashboard/components/ClickBehaviorSidebar/ActionOptions/ActionOptions.tsx

```diff
@@ -42,21 +47,21 @@ const mapDispatchToProps = {
 };
 
 function ActionOptions({
-  actions,
   dashcard,
   parameters,
   onUpdateButtonActionMapping,
-}: ActionOptionsProps & { actions: WritebackAction[] }) {
-  const connectedActionId = dashcard.action_id;
-
-  const selectedAction = actions.find(
-    action => action.id === connectedActionId,
-  );
+}: ActionOptionsProps) {
+  const selectedAction = dashcard.action;
 
   const handleActionSelected = useCallback(
     (action: WritebackAction) => {
       onUpdateButtonActionMapping(dashcard.id, {
-        action_id: action.id,
+        card_id: action.model_id,
+        action,
+        visualization_settings: {
+          ...dashcard.visualization_settings,
+          action_slug: action.slug, // :-( so hacky
```

Comment:
This is bad. This is clearly not a visualization setting, but this is kind of an intermediate backend requirement until we totally solidify on using actions-in-models. This will likely become a new column on the actions table down the road.

Reply:
Thanks for explaining what's going on here 🙏 Should we write this down somewhere?

Reply:
This is pretty well laid out in the Model-Actions planning doc: https://www.notion.so/metabase/Data-Apps-e372d425f81342d79a55387f9201ec62?p=5f038c453fbc43f192f2ba2bd4979069&pm=s